ipaserver/install/ipa_migrate.py

```diff
@@ -0,0 +1,2028 @@
+        if dns:
+            return val
+
+        # Replace host
+        if self.args.hostname in val:
+            val = val.replace(self.args.hostname, self.local_hostname)
+
+        # Replace domain
+        if self.remote_domain in val and self.local_domain not in val:
+            val = val.replace(self.remote_domain, self.local_domain)
+
+        # Replace realm
+        val = val.replace(self.remote_realm, self.realm)
+
+        return val
+
+    def convert_values(self, values, dns=False):
+        """
+        Replace suffix, hostname, domain, and realm in a list
+        """
+        new_values = []
+        for val in values:
+            new_values.append(self.convert_value(val, dns))
+
+        # normalize DN values
+        return self.normalize_vals(new_values)
+
+    def get_ldapentry_attr_vals(self, entry, attr):
+        """
+        Get the raw attribute values from IPA's LDAPEntry
+        """
+        vals = []
+        attr_vals = entry.raw.get(attr)
+        for val in attr_vals:
+            if isinstance(val, bytes):
+                vals.append(ensure_str(val))
+            elif not isinstance(val, str):
+                val = str(val)
+                vals.append(ensure_str(val))
+            else:
+                # Just a string
+                vals.append(val)
+
+        return self.normalize_vals(vals)
+
+    def build_ldap_entry(self, dn, attrs):
+        """
+        Take a DN and some attributes and build an LDAPEntry.  Used when
+        adding entries to the local server
+        """
+        entry = LDAPEntry(self.local_conn, DN(dn))
+        range_reset = False
+        for attr, values in attrs.items():
+            if (self.args.reset_range or self.mode == "stage-mode") and \
+                    attr.lower() in DNA_REGEN_ATTRS:
+                # Set the magic regen value
+                values = [DNA_REGEN_VAL]
+                self.log_debug(f"Resetting DNA range for new entry: {dn}")
+                range_reset = True
+            entry[attr] = values
+        if range_reset:
+            stats['reset_range'] += 1
+        return entry
+
+    def clean_entry(self, entry_dn, entry_type, entry_attrs):
+        """
+        Clean up the entry from the remote server
+
+        - Remove attributes from the ignore/strip list
+        - Reset suffix in all attributes
+        - If REALM was changed reset it to the new value
+        - Remove objectclasses from groups (IGNORE_OC list)
+        - userCertificate is removed if issued by IPA
+        - Remove unused objectclasses
+        """
+
+        # Don't clean DNS entries
+        if entry_type.startswith("dns"):
+            return entry_attrs
+
+        # Set the attrs we want to remove
+        remove_list = []
+        remove_attrs = STRIP_ATTRS + STRIP_OP_ATTRS
+        if self.args.mode != "prod-mode":
+            remove_attrs += PROD_ATTRS
+        if entry_type == "user":
+            remove_attrs.remove('ipasshpubkey')
```

Comment:
Why remove the ssh key from users?

Reply:
> Why remove the ssh key from users?

I thought you told me to :-)  But looking at the original design doc I don't see that being mentioned.  I'll remove this code